flight/Modules/OnScreenDisplay/onscreendisplay.c

```diff
@@ -0,0 +1,1448 @@
+ */
+
+// ****************
+
+// #define DEBUG_TIMING
+// #define DEBUG_ALARMS
+// #define DEBUG_TELEMETRY
+// #define DEBUG_BLACK_WHITE
+// #define DEBUG_STUFF
+#define SIMULATE_DATA
+#define TEMP_GPS_STATUS_WORKAROUND
+
+// static assert
+#define ASSERT_CONCAT_(a, b) a##b
+#define ASSERT_CONCAT(a, b) ASSERT_CONCAT_(a, b)
+/* These can't be used after statements in c89. */
+#ifdef __COUNTER__
+  #define STATIC_ASSERT(e,m) \
+    { enum { ASSERT_CONCAT(static_assert_, __COUNTER__) = 1/(!!(e)) }; }
+#else
+  /* This can't be used twice on the same line so ensure if using in headers
+   * that the headers are not included twice (by wrapping in #ifndef...#endif)
+   * Note it doesn't cause an issue when used on same line of separate modules
+   * compiled with gcc -combine -fwhole-program.  */
+  #define STATIC_ASSERT(e,m) \
+    { enum { ASSERT_CONCAT(assert_line_, __LINE__) = 1/(!!(e)) }; }
+#endif
+
+#include <openpilot.h>
+#include <pios_board_info.h>
+#include "pios_thread.h"
+#include "pios_semaphore.h"
+#include "misc_math.h"
+
+#include "onscreendisplay.h"
+#include "onscreendisplaysettings.h"
+#include "onscreendisplaypagesettings.h"
+#include "onscreendisplaypagesettings2.h"
+#include "onscreendisplaypagesettings3.h"
+#include "onscreendisplaypagesettings4.h"
+
+#include "modulesettings.h"
+#include "pios_video.h"
+
+#include "physical_constants.h"
+
+#include "accels.h"
+#include "accessorydesired.h"
+#include "airspeedactual.h"
+#include "attitudeactual.h"
+#include "baroaltitude.h"
+#include "flightstatus.h"
+#include "flightstatus.h"
+#include "flightbatterystate.h"
+#include "gpsposition.h"
+#include "positionactual.h"
+#include "gpstime.h"
+#include "gpssatellites.h"
+#include "gpsvelocity.h"
+#include "homelocation.h"
+#include "manualcontrolcommand.h"
+#include "systemalarms.h"
+#include "systemstats.h"
+#include "tabletinfo.h"
+#include "taskinfo.h"
+#include "velocityactual.h"
+#include "waypoint.h"
+#include "waypointactive.h"
+
+#include "osd_utils.h"
+#include "fonts.h"
+#include "font12x18.h"
+#include "font8x10.h"
+#include "WMMInternal.h"
+#include "logos.h"
+#include "mgrs.h"
+
+extern uint8_t PIOS_Board_Revision(void);
+
+extern uint8_t *draw_buffer_level;
+extern uint8_t *draw_buffer_mask;
+extern uint8_t *disp_buffer_level;
+extern uint8_t *disp_buffer_mask;
+
+// ****************
+// Private functions
+static void onScreenDisplayTask(void *parameters);
+
+// ****************
+// Private constants
+#define LONG_TIME        0xffff
+#define STACK_SIZE_BYTES 2048
+#define TASK_PRIORITY    PIOS_THREAD_PRIO_LOW
+#define UPDATE_PERIOD    100
+#define BLINK_INTERVAL_FRAMES 12
+#define BOOT_DISPLAY_TIME_MS (10*1000)
+
+const char METRIC_DIST_UNIT_LONG[] = "km";
+const char METRIC_DIST_UNIT_SHORT[] = "m";
+
+const char IMPERIAL_DIST_UNIT_LONG[] = "M";
+const char IMPERIAL_DIST_UNIT_SHORT[] = "ft";
+
+// Unit conversion constants
+#define MS_TO_KMH 3.6f
+#define MS_TO_MPH 2.23694f
+#define M_TO_FEET 3.28084f
```

Comment:
Put these into `physical_constants.h`. (Except `MS_TO_KMH` which is already defined there.)